File: Codingtest/Programmers/Javascript/Solution_68644/Solution_68644.js
function solution(numbers) {
  const NUM_LEN = numbers.length;

  let set = new Set();

  for (let i = 0; i < NUM_LEN; i++) {
    for (let j = i + 1; j < numbers.length; j++) {
      let a = parseInt(numbers[i] + numbers[j]);
      set.add(a);
    }
  }

  return Array.from(set).sort((a, b) => {
    return a - b;
  });
}

const numbers = [5, 0, 2, 7];
solution(numbers);

File: Codingtest/Programmers/JavaIntelliJ/src/Solution_68644.java
import java.util.ArrayList;
import java.util.Set;
import java.util.TreeSet;
import java.util.stream.Stream;

public class Solution_68644 {
    public static int[] solution(int[] numbers) {
//        int[] answer = new int[10000];
        ArrayList<Integer> answer = new ArrayList<Integer>();
        int NUM_LEN = numbers.length;
        TreeSet<Integer> ts = new TreeSet<Integer>();

        for(int i=0; i<NUM_LEN; i++) {
            for(int j=i+1; j<NUM_LEN; j++) {
                ts.add(numbers[i]+numbers[j]);
            }
        }

        for(Integer i : ts) {
            answer.add(i);
        }

        return answer.stream().mapToInt(i -> i).toArray();
    }

    public static void main(String[] args) {
        int[] nums = {2,1,3,4,1};

        int[] result = solution(nums);

        System.out.println("- result -");
        for (int data: result
             ) {
            System.out.println(data);
        }
    }
}
